In [1]:
%matplotlib inline
import pandas as pd

In [2]:
from IPython.core.display import HTML
css = open('style-table.css').read() + open('style-notebook.css').read()
HTML('<style>{}</style>'.format(css))

In [8]:
titles = pd.read_csv('data/titles.csv')
titles.head()

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:

d = pd.read_csv('F:/chrome downloads/Python/jupyter notebook files for Github/day 16-17/titles.csv')
d

title  year
0                        Tasveer Mere Sanam  1996
1                                  Only You  1994
2                      El pueblo del terror  1970
3                                   Machine  2007
4                                      MARy  2008
...                                     ...   ...
244909  Black Butterfly in a Colorful World  2018
244910        Hua fei hua wu chun man cheng  1980
244911                 Nippon dabi katsukyu  1970
244912        Under Siege 2: Dark Territory  1995
244913            She Must Be Seeing Things  1987

[244914 rows x 2 columns]

### How many movies are listed in the titles dataframe?

In [11]:
d['title'].size

244914

>therefor their are 244914 movies listed in the titles df

### What are the earliest two films listed in the titles dataframe?

In [25]:
d.nsmallest(2,['year'])

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

## or

In [26]:
d.head(2)

title  year
0  Tasveer Mere Sanam  1996
1            Only You  1994

### How many movies have the title "Hamlet"?

In [16]:
len(d[d.title == "Hamlet"])


20

>there are 20 Movies having title Hamlet

### How many movies are titled "North by Northwest"?

In [27]:
len(d[d.title == 'North by Northwest'])

1

>only one movie is title as North by Northwest

### When was the first movie titled "Hamlet" made?

In [39]:
d[d.title == 'Hamlet'].min()

title    Hamlet
year       1910
dtype: object

### List all of the "Treasure Island" movies from earliest to most recent.

In [93]:
d.sort_values("year", axis = 0, ascending = True,inplace = True, na_position ='first')
d
d[d.title == 'Treasure Island']


title  year
55686   Treasure Island  1918
196514  Treasure Island  1920
141822  Treasure Island  1934
77354   Treasure Island  1950
65051   Treasure Island  1972
37238   Treasure Island  1973
173051  Treasure Island  1985
27076   Treasure Island  1999

### How many movies were made in the year 1950?

In [3]:
len(d[d.year == 1950 ])

1113

### How many movies were made in the year 1960?

In [4]:
len(d[d.year == 1960 ])

1549

### How many movies were made from 1950 through 1959?

In [29]:
x=int(input("enter any year"))        
y=int(input("enter any year")) 
for i in range(x,y):
    a = len(d[d.year == i ])

    print(a)
    

enter any year1950
enter any year1959
1113
1165
1159
1210
1235
1262
1341
1437
1497


In [24]:
len(d[d.year == 1950 ])

1113

### In what years has a movie titled "Batman" been released?

In [8]:
d[d.title == 'Batman'].year

39349     1943
240896    1989
Name: year, dtype: int64

>the movie Batman was released in year 1943 and 1989

### How many roles were there in the movie "Inception"?

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

### But how many roles in the movie "Inception" did receive an "n" value?

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

### How many roles were credited in the silent 1921 version of Hamlet?

### How many roles were credited in Branagh’s 1996 Hamlet?

### How many "Hamlet" roles have been listed in all film credits through history?

### How many people have played an "Ophelia"?

### How many people have played a role called "The Dude"?

### How many people have played a role called "The Stranger"?

### How many roles has Sidney Poitier played throughout his career?

### How many roles has Judi Dench played?

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

### List the leading roles that Cary Grant played in the 1940s in order by year.

### How many roles were available for actors in the 1950s?

### How many roles were available for actresses in the 1950s?

### How many leading roles (n=1) were available from the beginning of film history through 1980?

### How many non-leading roles were available through from the beginning of film history through 1980?

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?